### PR TITLE
Use non-perl grep for version number extraction

### PR DESCRIPTION
**Pull request by brian-level**
Mac platform is becoming more locked down and grep installed from XCode for example is BSD grep not GNU grep so "-P" doesn't work.

This changes the grep in check-api-version script to extract the version numbers without needing perl patterns.

Also, on Mac platform, the [[ ]] syntax for checking the line count in the version file doesn't work because "wc -" prepends spaces on the count and [[ leaves spaces in.   Using the [ ] syntax works for Mac and other platforms as well but doesn't pass shellcheck, so manually stripping spaces is used.

### CODE REVIEW

File: script/check-api-version [hunk 1]
```diff
@@ -54,12 +54,15 @@ main()
...
-    [[ $(wc -l <"${OT_VERSIONS_FILE}") == 2 ]] || die 'Multiple OPENTHREAD_API_VERSION definitions found!'
+    line_count="$(wc -l <${OT_VERSIONS_FILE})"
+    line_count="${line_count// /}"
 
-    old_version=$(grep -aoP '(?<=-#define OPENTHREAD_API_VERSION ).+' "${OT_VERSIONS_FILE}") || die 'Failed to find old version!'
-    new_version=$(grep -aoP '(?<=\+#define OPENTHREAD_API_VERSION ).+' "${OT_VERSIONS_FILE}") || die 'Failed to find new version!'
+    [[ $line_count == 2 ]] || die 'Multiple OPENTHREAD_API_VERSION definitions found!'
+
+    old_version=$(grep '\-\#define OPENTHREAD_API_VERSION' "${OT_VERSIONS_FILE}" | grep -o '[0-9]..') || die 'Failed to find old version!'
+    new_version=$(grep '\+\#define OPENTHREAD_API_VERSION' "${OT_VERSIONS_FILE}" | grep -o '[0-9]..') || die 'Failed to find new version!'
```
Comment (bukepo) [inline script/check-api-version:65]: OpenThread API version is not fixed to 3 digits.

```suggestion
    old_version=$(grep '\-\#define OPENTHREAD_API_VERSION' "${OT_VERSIONS_FILE}" | grep -o '[1-9][0-9]*') || die 'Failed to find old version!'
    new_version=$(grep '\+\#define OPENTHREAD_API_VERSION' "${OT_VERSIONS_FILE}" | grep -o '[1-9][0-9]*') || die 'Failed to find new version!'
```

File: script/check-api-version [hunk 2]
```diff
@@ -54,12 +54,15 @@ main()
...
-    git diff "${OT_SHA_OLD}" -- include/openthread | tee >(cat >&2) | grep -aP '[-+]#define OPENTHREAD_API_VERSION (.+)' >"${OT_VERSIONS_FILE}" || die 'Version number is not updated!'
+    git diff "${OT_SHA_OLD}" -- include/openthread | tee >(cat >&2) | grep '[-+]#define OPENTHREAD_API_VERSION' >"${OT_VERSIONS_FILE}" || die 'Version number is not updated!'
 
-    [[ $(wc -l <"${OT_VERSIONS_FILE}") == 2 ]] || die 'Multiple OPENTHREAD_API_VERSION definitions found!'
+    line_count="$(wc -l <${OT_VERSIONS_FILE})"
+    line_count="${line_count// /}"
 
-    old_version=$(grep -aoP '(?<=-#define OPENTHREAD_API_VERSION ).+' "${OT_VERSIONS_FILE}") || die 'Failed to find old version!'
-    new_version=$(grep -aoP '(?<=\+#define OPENTHREAD_API_VERSION ).+' "${OT_VERSIONS_FILE}") || die 'Failed to find new version!'
+    [[ $line_count == 2 ]] || die 'Multiple OPENTHREAD_API_VERSION definitions found!'
```
Comment (bukepo) [inline script/check-api-version:62]:
```suggestion
```

File: script/check-api-version [hunk 3]
```diff
@@ -54,12 +54,15 @@ main()
         exit 0
     fi
 
-    git diff "${OT_SHA_OLD}" -- include/openthread | tee >(cat >&2) | grep -aP '[-+]#define OPENTHREAD_API_VERSION (.+)' >"${OT_VERSIONS_FILE}" || die 'Version number is not updated!'
+    git diff "${OT_SHA_OLD}" -- include/openthread | tee >(cat >&2) | grep '[-+]#define OPENTHREAD_API_VERSION' >"${OT_VERSIONS_FILE}" || die 'Version number is not updated!'
 
-    [[ $(wc -l <"${OT_VERSIONS_FILE}") == 2 ]] || die 'Multiple OPENTHREAD_API_VERSION definitions found!'
+    line_count="$(wc -l <${OT_VERSIONS_FILE})"
+    line_count="${line_count// /}"
```
Comment (bukepo) [inline script/check-api-version:60]: simpler?
```suggestion
    [[ $(($(wc -l <"${OT_VERSIONS_FILE}"))) == 2 ]] || die 'Multiple OPENTHREAD_API_VERSION definitions found!'
```

**Comment by jwhui** [comment]
@brian-level , any thoughts on the comments above?

File: script/check-api-version (hunk 3 above)
Comment (brian-level) [reply script/check-api-version:60]: yes, but the `[[` notation isn't portable and is why I did it in multiple steps

File: script/check-api-version (hunk 1 above)
Comment (brian-level) [reply script/check-api-version:65]: ok, that looks bette

**Comment by jwhui** [comment]
Closing stale PR. @brian-level , feel free to reopen this PR if you'd like to move it forward.

**Pull request closed without merging**